sdk/src/shortvec.rs

```diff
@@ -1,373 +1,37 @@
-use bincode::{deserialize_from, serialize_into, Error};
+use serde::de::Deserializer;
+use serde::ser::Serializer;
 use serde::Serialize;
-use std::io::{Cursor, Read, Write};
-use std::mem::size_of;
 
-pub fn encode_len<W: Write>(writer: &mut W, len: usize) -> Result<(), Error> {
-    let mut rem_len = len;
+// Use 1 to 9 bytes to store the length of the vector.
+pub fn serialize<S: Serializer, T: Serialize>(
+    elements: &[T],
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    use serde::ser::SerializeTuple;
+    let mut seq = serializer.serialize_tuple(0)?;
```

Comment:
@sambley, that zero is documented as a length. Not sure what it's supposed to be used for, but seems to be ignored by bincode.

Reply:
I read the bincode implementation. That parameter is ignored.